### PR TITLE
Update parsing to correctly parse file content with `++` or `--`

### LINKED ISSUE #41: Wrong result when paring some diff that contains deleted sql comment starting with `--`
Here is a test diff input to reproduce the bug
```
diff --git a/test.sql b/test.sql
index 305feaa..70865e7 100644
--- a/test.sql
+++ b/test.sql
@@ -1,10 +1,9 @@
--- Person
-create table person
-( id                   		bigint,
-  primary key ( id ) );
-
-
 -- Photo
 create table photo
 ( id                      bigint,
   primary key (id ) );
+
+-- Person
+create table person
+( id                   		bigint,
+  primary key ( id ) );
```
And the parsed result is
```
[
  {
    "chunks": [
      {
        "content": "@@ -1,10 +1,9 @@",
        "changes": [
          {
            "type": "del",
            "del": true,
            "ln": 1,
            "content": "-create table person"
          },
          {
            "type": "del",
            "del": true,
            "ln": 2,
            "content": "-( id                   \t\tbigint,"
          },
          {
            "type": "del",
            "del": true,
            "ln": 3,
            "content": "-  primary key ( id ) );"
          },
          {
            "type": "del",
            "del": true,
            "ln": 4,
            "content": "-"
          },
          {
            "type": "del",
            "del": true,
            "ln": 5,
            "content": "-"
          },
          {
            "type": "normal",
            "normal": true,
            "ln1": 6,
            "ln2": 1,
            "content": " -- Photo"
          },
          {
            "type": "normal",
            "normal": true,
            "ln1": 7,
            "ln2": 2,
            "content": " create table photo"
          },
          {
            "type": "normal",
            "normal": true,
            "ln1": 8,
            "ln2": 3,
            "content": " ( id                      bigint,"
          },
          {
            "type": "normal",
            "normal": true,
            "ln1": 9,
            "ln2": 4,
            "content": "   primary key (id ) );"
          },
          {
            "type": "add",
            "add": true,
            "ln": 5,
            "content": "+"
          },
          {
            "type": "add",
            "add": true,
            "ln": 6,
            "content": "+-- Person"
          },
          {
            "type": "add",
            "add": true,
            "ln": 7,
            "content": "+create table person"
          },
          {
            "type": "add",
            "add": true,
            "ln": 8,
            "content": "+( id                   \t\tbigint,"
          },
          {
            "type": "add",
            "add": true,
            "ln": 9,
            "content": "+  primary key ( id ) );"
          }
        ],
        "oldStart": 1,
        "oldLines": 10,
        "newStart": 1,
        "newLines": 9
      }
    ],
    "deletions": 0,
    "additions": 0,
    "from": "test.sql",
    "to": "test.sql",
    "index": [
      "305feaa..70865e7",
      "100644"
    ]
  },
  {
    "chunks": [],
    "deletions": 5,
    "additions": 5,
    "from": "Person"
  }
]
```
which is wrong.

### PR DESCRIPTION
**Pull request by mironiasty**
I had to update how files are parsing - and I made parser a little 'smarter'.

So right now, when chunk (`@@ x,y x,y@@`) is spotted, parser switch into 'parsing content` state. So it only check rules for files.

Fixes #41

### CODE REVIEW

File: __tests__/parse.spec.js [hunk 1]
```diff
@@ -163,14 +163,14 @@ diff --git a/file1 b/file1
 index 123..456 789
 --- a/file1
 +++ b/file1
-@@ -1,2 +1,2 @@
+@@ -1,1 +1,1 @@
```
Comment (mironiasty) [inline __tests__/parse.spec.js:166]: This and next one tests had invalid values here - diff contained only single lines added/removed.

File: __tests__/parse.spec.js [hunk 2]
```diff
@@ -230,7 +230,7 @@ index 123..456 789
 +
  Therefore let there always be non-being,
 	so we may see their subtlety,
-And let there always be being,
+ And let there always be being,
```
Comment (mironiasty) [inline __tests__/parse.spec.js:233]: I think space was missing here, not sure if that was intentional.

**Review by sergeyt — APPROVED** [review]

**Comment by mironiasty** [comment]
I just pushed updated/minified `indexj.js` file - so hopefully test will pass now.

**Review by sergeyt — APPROVED** [review]

File: __tests__/parse.spec.js (hunk 1 above)
Comment (sergeyt) [reply __tests__/parse.spec.js:166]: ok

File: __tests__/parse.spec.js (hunk 2 above)
Comment (sergeyt) [reply __tests__/parse.spec.js:233]: ok

**Comment by sergeyt** [comment]
@mironiasty thanks for the contribution, just released it

**Pull request merged**